### ISSUE TITLE
Add log level controls 

### ISSUE THREAD

**Issue by ahankinson** [rism-digital/verovio#3122]
Currently Verovio can enable or disable logging, but there are no other fine-grained controls for silencing specific log levels. So you cannot currently silence "Info" messages and only see "error" messages.

There are some global functions currently available for logging. These would need to be modified by including an enum to configure the logging. There should be four levels: error, warning, info, and debug. Each level includes the previous, so if you set "warning" you would get "warning" and "error"; if you choose "info" you would get warning, error, and info, etc.

In the JS toolkit we would need to expose a new global function to configure this behaviour. The built-in logging is already available in the C++ code (it just needs to be modified to support levels), so it needs to be exported in the various emscripten proxies, etc. Maybe @WolfgangDrescher has an idea of how this could work?

Related to this, I'm going to be submitting a PR to make a minor change to the JS toolkit to remove the `console.debug` statements there for the constructor / destructor. This should prevent the toolkit from emitting debug messages in production code.

**Comment by lpugin** [comment]
For the JS toolkit I added the functions to the C wrapper and exposed them in the emscripten build. They still needs to be made accessible from the module
https://github.com/rism-digital/verovio/commits/develop-js-logging
@WolfgangDrescher could you have a look? Thanks

**Comment by WolfgangDrescher** [comment]
> The built-in logging is already available in the C++ code (it just needs to be modified to support levels), so it needs to be exported in the various emscripten proxies, etc.

Are you speaking about the proxies in `emscripten/npm/src/emscripten-proxy.js`? lpugin already added a method there.
Will it be enough to make this function accessible via `VerovioToolkit`? Or is any kind of subscriber needed that listens to the logs of the module and then prints them in the console? Or will the emscripten module print them to the console automatically when `enableLog` is activated?

How would this logger be activated? E.g. like this?

```js
const LOG_LEVEL = 'debug';
verovio.module.onRuntimeInitialized = () => {
    const tk = new verovio.toolkit();
    tk.enableLog(LOG_LEVEL);
}
```

**Comment by ahankinson** [comment]
I think it might be something like:

```
const LOG_LEVEL = 'debug';
verovio.module.onRuntimeInitialized = () => {
    verovio.enableLog(LOG_LEVEL);
    const tk = new verovio.toolkit();
}
```

But maybe @lpugin can confirm?

**Comment by lpugin** [comment]
Yes, @ahankinson has it right. It is a module-level function. We need something more than what I added in the `emscripten-proxy.js` because when I call it I still get:
```js
Uncaught (in promise) TypeError: verovio.enableLog is not a function
```

**Comment by lpugin** [comment]
(PS As it is `enableLog` takes a bool)

**Comment by ahankinson** [comment]
> (PS As it is enableLog takes a bool)

How would you envision one with a log level? Maybe an enum: 

```
OFF = 0
DEBUG = 1
INFO = 2
WARNING = 3
ERROR = 4
```

then:

```
verovio.enableLog(verovio.INFO)
```

?

**Comment by WolfgangDrescher** [comment]
> I think it might be something like: `verovio.enableLog(LOG_LEVEL)`

This could get problematic for the ESM export since there is no global verovio instance. But something like `enableVerovioLog` or `enableVerovioModuleLog` could be exported by `emscripten/npm/src/index.js`. I would add `verovio` here into the naming so people don't get confused on what they are enabling the logs for.

> `verovio.enableLog(LOG_LEVEL);`

So this should be set per module and not per toolkit instance, correct? With the new ESM ready approach I think we should do something like this because there is no emscripten proxy without a toolkit right now:

```js
// verovio-logger.js
import { createEmscriptenProxy } from './emscripten-proxy.js';

export function enableVerovioLog(VerovioModule, level) {
    const proxy = createEmscriptenProxy(VerovioModule);
    proxy.enableLog(level);
}
```

> (PS As it is enableLog takes a bool)

In the `develop-js-logging` branch the function accepts a number? We can add dictionary to map the int/bool to the words. E.g.

```js
verovio.logLevel = {
   off/none: 0
   debug: 0
   info: 0
   warning: 0
   error: 0
}
```

then 

```js
// with verovio-toolkit-wasm.js
import verovio from 'verovio';
verovio.enableLog(verovio.logLevel.info);
// or to make it compatible with the ESM approach
verovio.enableLog(verovio.module, verovio.logLevel.info);

// with ESM
import { logLevel } from 'verovio/esm';
enableVerovioLog(VerovioModule, logLevel.info);
```

**Comment by WolfgangDrescher** [comment]
Btw. there is PSR 3 (PHP Standards Recommendations): https://www.php-fig.org/psr/psr-3/. We can use this as inspiration for the naming. So `verovio.LogLevel.ERROR` seems appropriate. Not sure about the uppercase in JS.

Maybe `enableLogger` instead of `enableLog`?

**Comment by lpugin** [comment]
We should keep `enableLog` because this is function we have in the other bindings / tools.

I am not sure what to do about the ESM issue since I am not familiar with the different JS approaches. If we can do
```
import verovio from 'verovio';
verovio.enableLog(verovio.logLevel.info);
```
that would be better.

**Comment by WolfgangDrescher** [comment]
Have a look at my pull request: https://github.com/rism-digital/verovio/pull/3125

**Comment by lpugin** [comment]
Implemented with #3122
